pinot-spi/src/main/java/org/apache/pinot/spi/env/SegmentMetadataPropertyReader.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.env;
+
+import com.google.common.base.Preconditions;
+import java.io.Reader;
+import java.util.List;
+import org.apache.commons.configuration2.PropertiesConfiguration.PropertiesReader;
+
+
+/**
+ * SegmentMetadataPropertyReader extends the PropertiesReader
+ * <p>
+ * Purpose: loads the segment metadata faster
+ *  - by skipping the unescaping of key and
+ *  - parsing the line by splitting based on first occurrence of separator
+ */
+class SegmentMetadataPropertyReader extends PropertiesReader {
+  private boolean _skipUnescapePropertyName;
+  private final String _segmentMetadataVersionHeader;
+
+  public SegmentMetadataPropertyReader(Reader reader, String segmentMetadataVersionHeader) {
+    super(reader);
+    _segmentMetadataVersionHeader = segmentMetadataVersionHeader;
+  }
+
+  @Override
+  protected void parseProperty(final String line) {
+    // if newer version of the segment metadata(based on version value in the property configuration header)
+    // skip the regex based parsing of the line content and splitting the content based on first occurrence of separator
+    if (!_skipUnescapePropertyName) {
```

Comment:
- **Explanation**: We override this method to set the `_skipUnescapePropertyName` flag. 
- **Assumption**: We will not set the version header for the old segment metadata. If we decide to add the header for old segment metadata, in that case we have to compare the version value as well. 

Header comment is the first content read from the config file, so it will set the flag value if `segment.metadata.version` header is present for the segment.

Reply:
for config file w/o this header, it seems like we'd have to enter this if-check for every property in the config file?

why not set check and set this flag in the constructor method?

Reply:
True, this if-check will run for every property if the header is not present in the segment metadata. We need to introduce some more conditions here.

Setting the flag value in the constructor wouldn't work if we want to set the value based on the header value present. Config with the header is determined when the PropertiesConfiguration reader reads the entire file content. Header is the first comment line, and it's set after reading the first property from the file([code reference](https://github.com/apache/commons-configuration/blob/master/src/main/java/org/apache/commons/configuration2/PropertiesConfigurationLayout.java#L672)).

Reply:
Added extra condition(`getCommentLines().size() > 0`) to skip some of the unnecessary check.